collector/aws/elasticsearch_test.go

```diff
@@ -0,0 +1,159 @@
+}
+
+func TestDescribeElasticSearchClusters(t *testing.T) {
+
+	collector := testutils.NewMockCollector()
+	metrics := []config.MetricConfig{}
+
+	t.Run("valid", func(t *testing.T) {
+
+		mockClient := MockAWSElasticSearchClient{
+			responseDescribeClusters: &defaultElasticSearchMock,
+		}
+
+		esManager := aws.NewElasticSearchManager(collector, &mockClient, nil, nil, nil, metrics, "us-east-1")
+
+		result, _ := esManager.DescribeClusters()
+
+		if len(result) != len(defaultElasticSearchMock.DomainStatusList) {
+			t.Fatalf("unexpected elasticsearch clusters count, got %d expected %d", len(result), len(defaultElasticSearchMock.DomainStatusList))
+		}
+	})
+
+	t.Run("error", func(t *testing.T) {
+
+		mockClient := MockAWSElasticSearchClient{
+			responseDescribeClusters: &defaultElasticSearchMock,
+			err:                      errors.New("error"),
+		}
+
+		esManager := aws.NewElasticSearchManager(collector, &mockClient, nil, nil, nil, metrics, "us-east-1")
+
+		_, err := esManager.DescribeClusters()
+
+		if err == nil {
+			t.Fatalf("unexpected describe clusters error, returned empty")
+		}
+	})
+
+}
+
+func TestDetectElasticSearch(t *testing.T) {
+
+	collector := testutils.NewMockCollector()
+	mockCloudwatchClient := MockAWSCloudwatchClient{
+		responseMetricStatistics: defaultResponseMetricStatistics,
+	}
+
+	mockSTSClient := MockAWSSTSClient{
+		responseGetCallerIdentity: &defaultSTSGetCallerIdentity,
+	}
+	stsManager := aws.NewSTSManager(&mockSTSClient)
+	cloutwatchManager := aws.NewCloudWatchManager(&mockCloudwatchClient)
+	pricingManager := aws.NewPricingManager(&defaultPricingMock, "us-east-1")
+
+	mockClient := MockAWSElasticSearchClient{
+		responseDescribeClusters: &defaultElasticSearchMock,
+	}
+
+	esManager := aws.NewElasticSearchManager(collector, &mockClient, cloutwatchManager, pricingManager, stsManager, defaultMetricConfig, "us-east-1")
+
+	response, _ := esManager.Detect()
+
+	if len(response) != 1 {
+		t.Fatalf("unexpected elasticsearch detected, got %d expected %d", len(response), 1)
+	}
+
+	if len(collector.Events) != 1 {
+		t.Fatalf("unexpected collector elasticsearch resources, got %d expected %d", len(collector.Events), 1)
+	}
+
+	if len(collector.EventsCollectionStatus) != 2 {
+		t.Fatalf("unexpected resource status events count, got %d expected %d", len(collector.EventsCollectionStatus), 2)
+	}
+
+}
+
+func TestDetectElasticSearchError(t *testing.T) {
+
+	collector := testutils.NewMockCollector()
+	mockCloudwatchClient := MockAWSCloudwatchClient{
+		responseMetricStatistics: defaultResponseMetricStatistics,
+	}
+	cloutwatchManager := aws.NewCloudWatchManager(&mockCloudwatchClient)
+	pricingManager := aws.NewPricingManager(&defaultPricingMock, "us-east-1")
+
+	mockClient := MockAWSElasticSearchClient{
+		err: errors.New(""),
+	}
+
+	esManager := aws.NewElasticSearchManager(collector, &mockClient, cloutwatchManager, pricingManager, nil, defaultMetricConfig, "us-east-1")
+
+	response, _ := esManager.Detect()
```

Comment:
`Detect` should't return error in this case?

Reply:
Added error and checked it.

Reply:
fixed.